scripts/HPC_chatbot_preprocessor/chatbot_parser.py

```diff
@@ -0,0 +1,593 @@
+    :return: the text to be written to the file, if any
+    """
+    # check whether the first part of the line contains information wrt if-statements
+    match = re.search(r'^\{-if-%(.*?)%-if-}(.*)', curr_line)
+
+    # check whether the line contains information wrt if-statements that is not in its first part
+    match_large = re.search(r'^(.*)(\{-if-%.*?%-if-})(.*)', curr_line)
+
+    if match:
+        content = match.group(1)
+
+        # new if-statement wrt OS with '=='
+        if re.search(r'if OS == ', content):
+            OS = content.split()[-1]
+
+            # set new active OS
+            active_OS_if_states[OS] = "active"
+
+            # set other active ones on inactive
+            for other_OS in active_OS_if_states.keys():
+                if other_OS != OS and active_OS_if_states[other_OS] == "active":
+                    active_OS_if_states[other_OS] = "inactive"
+
+        # new if-statement wrt OS with '!='
+        elif re.search(r'if OS != ', content):
+            OS = content.split()[-1]
+
+            # set new active OS
+            active_OS_if_states[OS] = "inactive"
+
+            # set other inactive ones on active
+            for other_OS in active_OS_if_states.keys():
+                if other_OS != OS and active_OS_if_states[other_OS] == "inactive":
+                    active_OS_if_states[other_OS] = "active"
+
+        # endif statement wrt OS
+        elif re.search(r'endif', content):
+            if str(1) in active_OS_if_states.values():
+                active_OS_if_states[
+                    list(active_OS_if_states.keys())[list(active_OS_if_states.values()).index(str(1))]] = "active"
+            else:
+                for key in active_OS_if_states.keys():
+                    active_OS_if_states[key] = "inactive"
+
+        # else statement wrt OS
+        elif re.search(r'else', content):
+
+            i = 0
+            for i in range(3):
+                if str(i) not in active_OS_if_states.values():
+                    break
+
+            # set the previously active one on inactive until the next endif
+            key_list = list(active_OS_if_states.keys())
+            position = list(active_OS_if_states.values()).index("active")
+            active_OS_if_states[key_list[position]] = str(i)
+
+            # set inactive ones on active
+            while "inactive" in active_OS_if_states.values():
+                position = list(active_OS_if_states.values()).index("inactive")
+                active_OS_if_states[key_list[position]] = "active"
+
+        if len(match.group(2)) != 0:
+            extra_message = match.group(2).lstrip()
+            return "check_extra_message", extra_message, None
+
+        else:
+            return "done", None, None
+
+    elif match_large:
+        return "write_text_and_check_extra_message", match_large.group(2), match_large.group(1)
+
+    else:
+        return "write_text", None, curr_line
```

Comment:
ACTION_WRITE_TEXT = "write_text"

```suggestion
        return ACTION_WRITE_TEXT, None, curr_line